contracts/dao/USDTLpListaDistributor.sol

```diff
@@ -0,0 +1,470 @@
+    address _pauser,
+    address _vault,
+    address _stakeVault
+  ) external initializer {
+    require(_admin != address(0), "admin is the zero address");
+    require(_manager != address(0), "manager is the zero address");
+    require(_vault != address(0), "vault is the zero address");
+    require(_stakeVault != address(0), "stake vault is the zero address");
+
+    __ReentrancyGuard_init();
+    __AccessControl_init();
+    __Pausable_init();
+
+    _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+    _setupRole(MANAGER, _manager);
+    _setupRole(VAULT, _vault);
+    _setupRole(PAUSER, _pauser);
+
+    lisUSD = IERC20(IStableSwap(stableSwapPool).coins(0));
+    usdt = IERC20(IStableSwap(stableSwapPool).coins(1));
+    lpToken = IStableSwap(stableSwapPool).token();
+
+    vault = IVault(_vault);
+    stakeVault = _stakeVault;
+    require(v2wrapper.rewardToken() == IStakingVault(stakeVault).rewardToken(), "Invalid reward token");
+    cake = IStakingVault(stakeVault).rewardToken();
+
+    harvestTimeGap = 1 hours;
+    isActive = true;
+
+    name = "USDT LP-Staked Reward Lista Distributor";
+    symbol = "USDTLpListaDistributor";
+  }
+
+  /* ================ External Functions ================ */
+
+  /**
+   * @dev Deposit USDT to PancakeStableSwapTwoPool and stake LP token to farming contract
+   * @param usdtAmount amount of USDT to deposit
+   * @param minLpAmount minimum amount of LP token required to mint
+   */
+  function deposit(uint256 usdtAmount, uint256 minLpAmount) external onlyActive {
+    require(usdtAmount > 0, "Invalid usdt amount");
+    uint256 expectLpAmount = getLpToMint(usdtAmount);
+    require(expectLpAmount >= minLpAmount, "Invalid min lp amount");
+
+    // 1. Transfer USDT to this contract
+    usdt.safeIncreaseAllowance(stableSwapPool, usdtAmount);
+    usdt.safeTransferFrom(msg.sender, address(this), usdtAmount);
+
+    // 2. Add USDT to PancakeStableSwapTwoPool
+    uint256 actualLpAmount = IERC20(lpToken).balanceOf(address(this));
+    IStableSwap(stableSwapPool).add_liquidity([0, usdtAmount], minLpAmount);
+    actualLpAmount = IERC20(lpToken).balanceOf(address(this)) - actualLpAmount;
+
+    require(actualLpAmount >= minLpAmount, "Invalid lp amount minted");
+
+    // 3. Update user's LP balance and LISTA reward, and distributor's LP total supply
+    _deposit(msg.sender, actualLpAmount);
+
+    // 4. Stake the received LP token to farming contract only if not in emergency mode
+    if (!emergencyMode) _stakeLp(msg.sender, actualLpAmount);
+
+    emit USDTStaked(lpToken, usdtAmount, actualLpAmount);
+  }
+
+  /**
+   * @dev Unstake LP token from farming contract and withdraw USDT and lisUSD from PancakeStableSwapTwoPool
+   * @param lpAmount amount of LP token
+   * @param minLisUSDAmount minimum amount of lisUSD to withdraw
+   * @param minUSDTAmount minimum amount of USDT to withdraw
+   */
+  function withdraw(uint256 lpAmount, uint256 minLisUSDAmount, uint256 minUSDTAmount) external {
```

Comment:
withdraw is allowed if onlyActive is not set?

Reply:
Yes, `isActive` is the switch for deposit only. Users are able to withdraw even if the distributor is inactive.